src/libraries/System.Runtime.InteropServices/gen/LibraryImportGenerator/Analyzers/CustomMarshallerAttributeAnalyzer.cs

```diff
@@ -632,56 +630,59 @@ public PerCompilationAnalyzer(Compilation compilation)
                 _spanOfT = compilation.GetBestTypeByMetadataName(TypeNames.System_Span_Metadata);
                 _readOnlySpanOfT = compilation.GetBestTypeByMetadataName(TypeNames.System_ReadOnlySpan_Metadata);
             }
-
-            public void AnalyzeAttribute(SyntaxNodeAnalysisContext context)
+            public void AnalyzeAttribute(OperationAnalysisContext context)
             {
-                AttributeSyntax syntax = (AttributeSyntax)context.Node;
-                ISymbol attributedSymbol = context.ContainingSymbol!;
-
-                AttributeData? attr = syntax.FindAttributeData(attributedSymbol);
-                if (attr?.AttributeClass?.ToDisplayString() == TypeNames.CustomMarshallerAttribute
-                    && attr.AttributeConstructor is not null)
+                IAttributeOperation attr = (IAttributeOperation)context.Operation;
+                if (attr.Operation is IObjectCreationOperation attrCreation
+                    && attrCreation.Type.ToDisplayString() == TypeNames.CustomMarshallerAttribute)
                 {
-                    DiagnosticReporter managedTypeReporter = DiagnosticReporter.CreateForLocation(syntax.FindArgumentWithNameOrArity("managedType", 0).FindTypeExpressionOrNullLocation(), context.ReportDiagnostic);
-                    INamedTypeSymbol entryType = (INamedTypeSymbol)attributedSymbol;
-
-                    ITypeSymbol? managedTypeInAttribute = (ITypeSymbol?)attr.ConstructorArguments[0].Value;
-                    if (managedTypeInAttribute is null)
+                    INamedTypeSymbol entryType = (INamedTypeSymbol)context.ContainingSymbol!;
+                    if (attrCreation.Arguments[0] is IArgumentOperation { Value: { ConstantValue: { HasValue: true, Value: null } } nullManagedTypeValue })
```

Comment:
Is the check for the null argument value worth pulling out into a helper? There's enough wrapped up in the check it takes a moment (and maybe some doc searching) to get the logic and this PR has it in a couple places already.